src/ListItem.js

```diff
@@ -3,16 +3,19 @@ import { buyItem } from './services/fetch-utils';
 export default function ListItem({ fetchItems, listItem }) {
   async function handleClick() {
     // buy the item (in supabase)
+    buyItem(listItem.id);
     // refetch the updated items array by calling the function passed in through props
+    fetchItems();
   }
 
 //  () => {} is javascript for "do nothing". It's an arrow function that doesn't nothing at all.
   return (
     // on click, if it's already been bought, do nothing; otherwise, call the handleClick function
-    <div className='list-item'>
+    <div className='list-item' onClick={listItem.has_been_bought ? () => {} : handleClick()}>
```

Comment:
I don't think you want to call `handleClick` here.  You are just making a declaration that `handleClick` should be called `onClick`.  This is causing an error in your console that is preventing your list from updating correctly on the page.